src/main/java/org/gridsuite/modification/server/modifications/GeneratorScaling.java

```diff
@@ -47,9 +46,11 @@ protected void applyStackingUpVariation(Network network,
                                          ScalingVariationInfos generatorScalingVariation) {
         AtomicReference<Double> sum = new AtomicReference<>(0D);
         Scalable stackingUpScalable = Scalable.stack(identifiableAttributes.stream()
-                .map(equipment -> {
-                    sum.set(network.getGenerator(equipment.getId()).getTargetP() + sum.get());
-                    return getScalable(equipment.getId());
+                .map(attribute -> network.getGenerator(attribute.getId()))
+                .filter(g -> g != null && g.getTerminal().isConnected())
```

Comment:
Maybe add a method in ModificationUtils:
public boolean isInjectionConnected(Injection<?> injection) {
        return injection != null && injection.getTerminal().isConnected();
    }
and you can call it everywhere

Reply:
yes, done